raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -0,0 +1,292 @@
+# -*- coding: utf-8 -*-
+from copy import deepcopy
+
+from raiden.transfer.architecture import Iteration
+from raiden.transfer.state import RoutesState
+from raiden.transfer.mediated_transfer.state import (
+    InitiatorState,
+    LockedTransferState,
+)
+from raiden.transfer.mediated_transfer.transition import update_route
+from raiden.transfer.state_change import (
+    ActionCancelTransfer,
+    ActionRouteChange,
+    Block,
+)
+from raiden.transfer.mediated_transfer.state_change import (
+    ActionCancelRoute,
+    ActionInitInitiator,
+    ReceiveSecretRequest,
+    ReceiveSecretReveal,
+    ReceiveTransferRefund,
+)
+from raiden.transfer.mediated_transfer.events import (
+    EventTransferFailed,
+    SendMediatedTransfer,
+    SendRevealSecret,
+)
+from raiden.utils import sha3
+
+
+def cancel_current_route(state):
+    """ Clear current state and try a new rotue.
+
+    - Discards the current secret
+    - Add the current route to the canceled list
+    - Add the current message to the canceled transfers
+    """
+    assert state.revealsecret is None, 'cannot cancel a transfer with a RevealSecret in flight'
```

Comment:
I see. And will we remove them in the future or just run with `python -O` in production?